examples/text-generation/aqlm_quantization/readme.md

```diff
@@ -0,0 +1,260 @@
+    └── Kompress
+        └── aqlm_quantization
+            ...
+```
+
+## Performance Evaluation
+
+After quantization, evaluate the performance of the quantized model using the provided evaluation script:
+
+```bash
+pip install lm-eval
+
+## ===== GSM8K Evaluation =====
+
+# baseline gsm8k 5 shot evaluation
+accelerate launch --no-python lm_eval --model hf \
+  --model_args pretrained=meta-llama/Meta-Llama-3-8B-Instruct,cache_dir=user_data/.cache \
+  --tasks gsm8k \
+  --num_fewshot 5 \
+  --batch_size "auto" \
+  --output_path user_data/evals/meta-llama_3.1-8b/base/gsm8k/
+
+# Llama3* gsm8k 5 shot evaluation
+accelerate launch --no-python lm_eval --model hf \
+  --model_args pretrained=RLHFlow/LLaMA3-iterative-DPO-final,cache_dir=user_data/.cache \
+  --tasks gsm8k \
+  --num_fewshot 5 \
+  --batch_size "auto" \
+  --output_path user_data/evals/meta-llama_3.1-8b/sft+iterative_dpo/gsm8k/
+
+# Llama3Q gsm8k 5 shot evaluation
+python examples/text-generation/aqlm_quantization/evaluate.py \
+  --base_model "meta-llama/Meta-Llama-3-8B-Instruct" \
+  --quantized_model "user_data/jobs/Kompress/aqlm_quantization/tmp/caliberation/" \
+  --tasks "gsm8k:5" \
+  --results "user_data/evals/meta-llama_3.1-8b/Llama3Q" \
+  --cache_dir "user_data/.cache"
+
+# Llama3Q PV Tuned gsm8k 5 shot evaluation
+python examples/text-generation/aqlm_quantization/evaluate.py \
+  --base_model "meta-llama/Meta-Llama-3-8B-Instruct" \
+  --quantized_model "user_data/jobs/Kompress/aqlm_quantization/tmp/converted/" \
+  --tasks "gsm8k:5" \
+  --results "user_data/evals/meta-llama_3.1-8b/Llama3Q_PV_Tuned" \
+  --cache_dir "user_data/.cache"
+
+## ===== Preplexity Evaluation =====
+
+# Llama3* perplexity evaluation
+python examples/text-generation/aqlm_quantization/evaluate.py \
+  --base_model "RLHFlow/LLaMA3-iterative-DPO-final" \
+  --tasks "gptq_wikitext:0" \
+  --results "user_data/evals/meta-llama_3.1-8b/Llama3*" \
+  --cache_dir "user_data/.cache"
+
+# baseline & Llama3Q perplexity evaluation
+python examples/text-generation/aqlm_quantization/evaluate.py \
+  --base_model "meta-llama/Meta-Llama-3-8B-Instruct" \
+  --quantized_model "user_data/jobs/Kompress/aqlm_quantization/tmp/caliberation/" \
+  --tasks "gptq_wikitext:0" \
+  --results "user_data/evals/meta-llama_3.1-8b" \
+  --cache_dir "user_data/.cache"
+
+# Llama3Q PV Tuned perplexity evaluation
+python examples/text-generation/aqlm_quantization/evaluate.py \
+  --base_model "meta-llama/Meta-Llama-3-8B-Instruct" \
+  --quantized_model "user_data/jobs/Kompress/aqlm_quantization/tmp/converted/" \
+  --tasks "gptq_wikitext:0" \
+  --results "user_data/evals/meta-llama_3.1-8b/Llama3Q_PV_Tuned" \
+  --cache_dir "user_data/.cache"
+```
+
+Compare the results with the original model to assess the impact of quantization on accuracy and inference speed.
+
+|                 | Llama3-8b  | Llama3Q PV Tuned  |
+|---------------- |----------- |------------------ |
+| GSM8K (5 shot)  | 50.9       | 58.9              |
```

Comment:
lets mention that we have used a fine-tuned llama3 and also include its number

Reply:
as above